src/middlewared/middlewared/plugins/nfs_/v4_freebsd.py

```diff
@@ -23,6 +23,31 @@ def setup_v4(self):
             subprocess.run(["service", "gssd", "forcestop"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
 
         if config["v4"]:
+            if self.middleware.call_sync("failover.licensed"):
+                # on HA systems, when using NFSv4, starting with (we believe) ESXi 7+,
+                # the ESXi NFS client expects that the nfs server scope or owner major
+                # string be the same between the controllers. If they mismatch, on failover
+                # the ESXi hypervisors go into a state that, apparently, requires them to be
+                # rebooted. This defeats the entire purpose of the "stateful"-ness of NFSv4.
+                # To prevent this scenario from occurring, we must set these values to a
+                # globally unique value between the controllers.
+                # NOTE: If this is changed while nfsd is started, then problems will occur
+                # so it's important that if you're reading this and decide to change it that
+                # you restart the nfsd service and then have your clients unmount and remount.
+                owner_major = config["v4_owner_major"]
+                if not owner_major:
+                    owner_major = str(uuid.uuid4())
+                    # write this to the database since it shouldn't change after it's been set
+                    # and has to be shared between controllers on HA systems
+                    self.middleware.call_sync(
+                        "datastore.update", "services.nfs", config["id"],
+                        {"v4_owner_major": owner_major},
+                        {"prefix": self._config.datastore_prefix},
+                    )
+
+                sysctl.filter("vfs.nfsd.scope")[0].value = owner_major
```

Comment:
you're assigning `owner_major` to both `scope` **and** `owner_major` ?

Reply:
Yeah they both need to be set to the same string. I'll update comments

Reply:
kewl -- yeah, comment should be good 👍🏼